Pytest tests/test_cli.py::test_accelerator_mismatch Broken

## Replication (on master)

`mv ~/.aws ~/.aws1` (just like on the Github machines)
`pytest tests/test_cli.py::test_accelerator_mismatch`

## Logs

```
file_path = '/var/folders/vf/41whbvcj47b2scn9slkfd8pm0000gn/T/tmpxwupyf25.yml', gpus = 'V100:1'

    def _capture_match_gpus_spec(file_path, gpus: str):
        result = cli_runner.invoke(cli.launch,
                                   [file_path, '--gpus', gpus, '--dryrun'])
>       assert not result.exit_code
E       assert not 1
E        +  where 1 = <Result ResourcesUnavailableError("Task Task(run=None)\n  resources: {AWS(p3.2xlarge, {'V100': 1})} requires AWS which is not enabled. Run `sky check` to enable access to it, or change the cloud requirement.")>.exit_code

tests/test_cli.py:62: AssertionError
```

It has to do with some modifications to `optimizer.py`

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -1,13 +1,11 @@
-import pytest
 import tempfile
 import textwrap
 
-import click
 from click import testing as cli_testing
 
 import sky
-import sky.cli as cli
 from sky import clouds
+import sky.cli as cli
 
 
 def test_infer_gpunode_type():
@@ -45,6 +43,13 @@ def test_infer_tpunode_type():
 
 def test_accelerator_mismatch(monkeypatch):
     """Test the specified accelerator does not match the instance_type."""
+    # Monkey-patching is required because in the test environment, no cloud is
+    # enabled. The optimizer checks the environment to find enabled clouds, and
+    # only generates plans within these clouds. The tests assume that all three
+    # clouds are enabled, so we monkeypatch the `sky.global_user_state` module
+    # to return all three clouds. We also monkeypatch `sky.check.check` so that
+    # when the optimizer tries calling it to update enabled_clouds, it does not
+    # raise exceptions.
     enabled_clouds = list(clouds.CLOUD_REGISTRY.values())
     monkeypatch.setattr(
         'sky.global_user_state.get_enabled_clouds',

--- a/tests/test_optimizer_dryruns.py
+++ b/tests/test_optimizer_dryruns.py
@@ -24,7 +24,7 @@ def _test_parse_accelerators(spec, expected_accelerators):
 # clouds are enabled, so we monkeypatch the `sky.global_user_state` module
 # to return all three clouds. We also monkeypatch `sky.check.check` so that
 # when the optimizer tries calling it to update enabled_clouds, it does not
-# raise SystemExit.
+# raise exceptions.
 def _make_resources(
     monkeypatch,
     *resources_args,

